### PR TITLE
Bugfix/xref syntax highlighting

### PR DESCRIPTION
**Pull request by nbfalcon**
When using xref to show all references/definitions/... of some symbol, all entries were gray, even though they had syntax highlighting in the buffer where they were defined. This was because `lsp--extract-line-from-buffer', used by lsp's xref routines, used
`buffer-substring-no-properties' to extract lines. Use `buffer-substring' instead.

If needed, I can rebase this against master.

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -480,32 +480,32 @@ the server has requested that."
 ;;;###autoload(put 'lsp-enable-file-watchers 'safe-local-variable #'booleanp)
 
 (defcustom lsp-file-watch-ignored '(; SCM tools
```
Comment (kiennq) [inline lsp-mode.el:482]: this changes is not related.

Comment (nbfalcon) [reply lsp-mode.el:482]: Yup, layered commits. All of my current lsp-mode PRs have this issue, except the first one.

**Comment by nbfalcon** [comment]
And that one is rebased.

File: lsp-mode.el (hunk 1 above)
Comment (nbfalcon) [reply lsp-mode.el:482]: Fixed it.

**Comment by nbfalcon** [comment]
I just removed the proper-regexes commits, so this now only contains a single trivial change, and I rebased against master.

**Comment by kiennq** [comment]
This is still not correctly rebased :(.
Your changes are consist of 2 commits which include unnecessary change.

**Comment by leungbk** [comment]
@nbfalcon The easiest way to handle this is to, on your current `xref` branch, go to `magit-status` and press `A d RET <select your master branch> RET` to donate this last commit to the master branch. Then, still on your `xref` branch, `O h` to reset your `xref` branch to your new master branch.

File: lsp-mode.el [hunk 2]
```diff
@@ -480,12 +480,12 @@ the server has requested that."
...
-                                    "[/\\\\]\\.git$"
```
Comment (seagle0128) [inline lsp-mode.el:483]: Question: why changed these regexp?

Comment (nbfalcon) [reply lsp-mode.el]: I resolved conflicts incorrectly.

**Comment by nbfalcon** [comment]
Now it is only a single SLOC commit - sorry for the trouble.

**Review by kiennq — APPROVED** [review]
